metric.go

```diff
@@ -136,27 +113,26 @@ func (m *MetricDefinition) NameWithTags() string {
 		return m.nameWithTags
 	}
 
-	sort.Strings(m.Tags)
+	nameWithTagsBuffer := &bytes.Buffer{}
+	writeSortedTagString(nameWithTagsBuffer, m.Name, m.Tags)
+	m.nameWithTags = nameWithTagsBuffer.String()
 
-	nameWithTagsBuffer := bytes.NewBufferString(m.Name)
-	tagPositions := make([]int, 0, len(m.Tags)*2)
+	var i int
+	cursor := len(m.Name)
+	m.Name = m.nameWithTags[:cursor]
 	for _, t := range m.Tags {
-		if len(t) >= 5 && t[:5] == "name=" {
+		if len(t) > 5 && t[:5] == "name=" {
 			continue
 		}
-
-		nameWithTagsBuffer.WriteString(";")
-		tagPositions = append(tagPositions, nameWithTagsBuffer.Len())
-		nameWithTagsBuffer.WriteString(t)
-		tagPositions = append(tagPositions, nameWithTagsBuffer.Len())
+		m.Tags[i] = m.nameWithTags[cursor+1 : cursor+1+len(t)]
+		cursor += len(t) + 1
+		i++
 	}
 
-	m.nameWithTags = nameWithTagsBuffer.String()
-	m.Tags = make([]string, len(tagPositions)/2)
-	for i := 0; i < len(m.Tags); i++ {
-		m.Tags[i] = m.nameWithTags[tagPositions[i*2]:tagPositions[i*2+1]]
+	// if a "name" tag existed, then we might have to shorten the slice
```

Comment:
it's not clear here why we are doing this. also the name tag may not be the last one in the list, so isn't it likely we're cutting off the wrong tag here?